listenbrainz/webserver/static/js/tests/personal-recommendations/PersonalRecommendationsModal.test.tsx

```diff
@@ -0,0 +1,254 @@
+};
+
+const globalProps = {
+  APIService: new APIServiceClass(""),
+  currentUser: user,
+  spotifyAuth: {},
+  youtubeAuth: {},
+};
+
+// Font Awesome generates a random hash ID for each icon everytime.
+// Mocking Math.random() fixes this
+// https://github.com/FortAwesome/react-fontawesome/issues/194#issuecomment-627235075
+jest.spyOn(global.Math, "random").mockImplementation(() => 0);
+
+describe("PersonalRecommendationModal", () => {
+  it("renders everything right", () => {
+    const wrapper = mount<PersonalRecommendationModal>(
+      <PersonalRecommendationModal
+        recordingToPersonallyRecommend={recordingToPersonallyRecommend}
+        newAlert={jest.fn()}
+      />
+    );
+
+    expect(wrapper.html()).toMatchSnapshot();
+  });
+});
+
+describe("submitPersonalRecommendation", () => {
+  it("calls API, and creates new alert on success", async () => {
+    const wrapper = mount<PersonalRecommendationModal>(
+      <GlobalAppContext.Provider value={globalProps}>
+        <PersonalRecommendationModal
+          recordingToPersonallyRecommend={recordingToPersonallyRecommend}
+          newAlert={jest.fn()}
+        />
+      </GlobalAppContext.Provider>
+    );
+
+    const instance = wrapper.instance();
+    const spy = jest.spyOn(
+      instance.context.APIService,
+      "submitPersonalRecommendation"
+    );
+    spy.mockImplementation((userToken, userName, metadata) => {
+      return Promise.resolve(200);
+    });
+
+    wrapper.setState({ blurbContent: "hii", users: ["riksucks", "hrik2001"] });
+    await instance.submitPersonalRecommendation();
+
+    expect(spy).toHaveBeenCalledTimes(1);
+    expect(spy).toHaveBeenCalledWith("auth_token", "name", {
+      recording_mbid: "recording_mbid",
+      recording_msid: "recording_msid",
+      artist_name: "TWICE",
+      track_name: "Feel Special",
+      release_name: undefined,
+      blurb_content: "hii",
+      users: ["riksucks", "hrik2001"],
+    });
+    expect(instance.props.newAlert).toHaveBeenCalledTimes(1);
+  });
+
+  it("blurbContent is reset after successful posting", async () => {
+    const wrapper = mount<PersonalRecommendationModal>(
+      <GlobalAppContext.Provider value={globalProps}>
+        <PersonalRecommendationModal
+          recordingToPersonallyRecommend={recordingToPersonallyRecommend}
+          newAlert={jest.fn()}
+        />
+      </GlobalAppContext.Provider>
+    );
+
+    const instance = wrapper.instance();
+    const spy = jest.spyOn(
+      instance.context.APIService,
+      "submitPersonalRecommendation"
+    );
+    spy.mockImplementation((userToken, userName, metadata) => {
+      return Promise.resolve(200);
+    });
+
+    wrapper.setState({ blurbContent: "hii", users: ["riksucks", "hrik2001"] });
+    const setStateSpy = jest.spyOn(instance, "setState");
+    await instance.submitPersonalRecommendation();
+    expect(setStateSpy).toHaveBeenCalledTimes(1);
+    expect(wrapper.state("blurbContent")).toEqual("");
```

Comment:
The "belt and braces" approach here is to first ensure that the state is indeed what you expect as a "normal" condition before you take an action.
In this case, it means adding `expect(wrapper.state("blurbContent")).toEqual("hii");` right after your setState call.